packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1578,6 +1572,7 @@ void ArbCore::operator()() {
                     core_machine->machine_state.output.arb_gas_used);
                 if (!status.ok()) {
                     core_error_string = status.ToString();
+                    core_error = true;
```

Comment:
I think it'd be cleaner to have a `setCoreError()` function that sets both these fields to make sure we always set them together

Reply:
:+1:

Reply:
:+1: